shell/probe/raindrop.ps1

```diff
@@ -8,11 +8,10 @@ function Execute {
         $stderrTempFile = New-Item -path "$dir\stderr.log" -Force
         $proc = Start-Process -FilePath $File -NoNewWindow -PassThru -RedirectStandardOutput $stdoutTempFile -RedirectStandardError $stderrTempFile
 
-        $errVar = $null
-        $proc | Wait-Process -Timeout 45 -ErrorAction SilentlyContinue -ErrorVariable errVar
+        $timeoutVar = $null
```

Comment:
do you even need this line? You said you tested and it worked when this line was still errVar, so maybe powershell defaults undeclared variables to null already?

Reply:
Yes, it's needed. Also that value exists again. 
```
while (1) {
 $proc = Start-Process -FilePath "notepad.exe" -NoNewWindow -PassThru
 if ($timeoutVar -eq $null) {
  Write-Host "is null"
 } else {
  Write-Host "is not null"
 }
 $proc | Wait-Process -Timeout 5 -ErrorAction SilentlyContinue -ErrorVariable timeoutVar
 Write-Host $timeoutVar
 $proc | kill
}
```

```
PS C:\Users\Administrator\Desktop> .\t.ps1
is null
This command stopped operation because process "notepad (7692)" is not stopped in the specified time-out.
is not null
This command stopped operation because process "notepad (444)" is not stopped in the specified time-out.
is not null
This command stopped operation because process "notepad (7324)" is not stopped in the specified time-out.
is not null
This command stopped operation because process "notepad (424)" is not stopped in the specified time-out.
is not null
This command stopped operation because process "notepad (7424)" is not stopped in the specified time-out.
```

Reply:
the while loop is outside this function, and hopefully powershell isnt so crazy as to keep that variable around between Executes

this is small potatoes, i'm just looking for ways to cut down chars

Reply:
Yeah, you're right 
```
function Execute {
 $proc = Start-Process -FilePath "notepad.exe" -NoNewWindow -PassThru
 if ($timeoutVar -eq $null) {
  Write-Host "is null"
 } else {
  Write-Host "is not null"
 }
 $proc | Wait-Process -Timeout 5 -ErrorAction SilentlyContinue -ErrorVariable timeoutVar
 Write-Host $timeoutVar
 $proc | kill
}


while (1) {
 Execute
}
```

```
.\t.ps1                                                                              is null                                                                                                                 This command stopped operation because process "notepad (5840)" is not stopped in the specified time-out.               is null                                                                                                                 This command stopped operation because process "notepad (6164)" is not stopped in the specified time-out.               is null                                                                                                                 This command stopped operation because process "notepad (6072)" is not stopped in the specified time-out.               is null
This command stopped operation because process "notepad (10232)" is not stopped in the specified time-out.
```